frontend/store/network.ts

```diff
@@ -60,31 +64,59 @@ export const getters = {
         }
         return getters.getPageNetwork;
     },
+    isChangingNetwork(state: State) {
+        return state.isChangingNetwork;
+    },
 };
 
 export const mutations = {
     setWalletChainId(state: State, walletChainId: string): void {
         state.walletChainId = walletChainId;
     },
+    setIsChangingNetwork(state: State, isChangingNetwork: boolean): void {
+        state.isChangingNetwork = isChangingNetwork;
+    },
 };
 
 export const actions = {
     setWalletChainId({ commit }: ActionContext<State, State>, walletChainId: string): void {
         commit('setWalletChainId', walletChainId);
     },
     async setPageNetwork(
-        { getters, dispatch, rootState }: ActionContext<State, any>,
+        { getters, dispatch, commit, rootState }: ActionContext<State, any>,
         newNetwork: string
     ): Promise<void> {
         const oldNetwork = getters.getPageNetwork;
         if (newNetwork === oldNetwork) {
             return;
         }
+        commit('setIsChangingNetwork', true);
         try {
             await dispatch('setWalletNetwork', newNetwork);
-            window.location.href = `${rootState.route.path}?network=${newNetwork}`;
+            await window.$nuxt.$router.replace({
+                path: rootState.route.path,
+                query: {
+                    network: newNetwork,
+                },
+            });
+            if (networkChangeTimeoutId) {
+                clearTimeout(networkChangeTimeoutId);
+            }
+
+            networkChangeTimeoutId = setTimeout(() => {
+                if (getters.getMakerNetwork === undefined) {
+                    message.error(`Failed to change network to ${newNetwork}`);
+                    commit('setIsChangingNetwork', false);
+                    window.$nuxt.$router.replace({
+                        query: {
+                            network: oldNetwork,
+                        },
+                    });
+                }
```

Comment:
> Please check if you need to clear networkChangeTimeoutId in some other actions (not sure about this one)

Nice call! After testing some things, I've found that switching the network, and then, while the timeout is waiting, switching it again but with the MetaMask extension, the timeout doesn't get, reset leading to unwanted behavior. 

**reference**

![2022-06-21 12 41 26](https://user-images.githubusercontent.com/47861849/174781551-59230a45-ea7f-4287-bb4c-80679dcf8d3d.gif)

I think also clearing it in the `setup` action is the best way to fix this so that whenever the `walletChainId` changed, the timeout gets cleared.